devices/stm32f429xx/stm32f429xx_usart.c

```diff
@@ -0,0 +1,597 @@
+#define GPIO_OSPEEDR_50M                (uint32_t) (0x2)                    /* Output speed 50MHz */
+    uint32_t speed = GPIO_OSPEEDR_50M;
+    my_GPIO->OSPEEDR &= ~(GPIO_OSPEEDR_M(pin_tx));
+    my_GPIO->OSPEEDR |= (speed << GPIO_OSPEEDR_PIN(pin_tx));
+    my_GPIO->OSPEEDR &= ~(GPIO_OSPEEDR_M(pin_rx));
+    my_GPIO->OSPEEDR |= (speed << GPIO_OSPEEDR_PIN(pin_rx));
+}
+
+
+static void enableUartToInterruptOnReceivedChar(uint32_t index)
+{
+    __mriStm32f429xxState.pCurrentUart->pUartRegisters->CR1 |= USART_CR1_RXNEIE;
+}
+
+
+static void setManualBaudFlag(void)
+{
+    __mriStm32f429xxState.flags |= STM32F429XX_UART_FLAGS_MANUAL_BAUD;
+}
+
+//If the order of g_uartConfigurations changes,fix it!
+/*  Return 0 for USART1
+ *  Return 1 for USART2
+ *  Return 2 for USART3
+ */
+int Platform_CommUartIndex(void)
+{
+    return __mriStm32f429xxState.pCurrentUart - g_uartConfigurations;
+}
+
+uint32_t Platform_CommHasReceiveData(void)
+{
+    return __mriStm32f429xxState.pCurrentUart->pUartRegisters->SR & USART_SR_RXNE;
+}
+
+int Platform_CommReceiveChar(void)
+{
+    while(!Platform_CommHasReceiveData()) 
+	{
+        //busy wait
+    }
+    return (__mriStm32f429xxState.pCurrentUart->pUartRegisters->DR & 0x1FF);
+}
+
+
+
+void Platform_CommSendChar(int Character)
+{
+    USART_TypeDef *uart = __mriStm32f429xxState.pCurrentUart->pUartRegisters;
+    while (!(uart->SR & USART_SR_TXE)) 
+	{
+        //busy wait
+    }
+    uart->DR = (Character & 0x1FF);
+}
+
+int Platform_CommCausedInterrupt(void)
+{
+    int interruptSource = (int)getCurrentlyExecutingExceptionNumber()-16;
+
+    //For USART1~3,the IRQn are continuous,others need check!
+    int irq_num_base = USART1_IRQn;
+    int currentUartIRQ = irq_num_base + Platform_CommUartIndex();
+    return currentUartIRQ==interruptSource;
+}
+
+void Platform_CommClearInterrupt(void)
+{
+    //Clear Interrupt flag,to avoid infinit loop in USARTx_Handler
+    __mriStm32f429xxState.pCurrentUart->pUartRegisters->SR &= ~USART_SR_RXNE;
+}
+
+int Platform_CommSharingWithApplication(void)
+{
+    return __mriStm32f429xxState.flags & STM32F429XX_UART_FLAGS_SHARE;
+}
+
+static int isManualBaudRate(void)
+{
+    return (int)(__mriStm32f429xxState.flags & STM32F429XX_UART_FLAGS_MANUAL_BAUD);
+}
+
+int Platform_CommShouldWaitForGdbConnect(void)
+{
+    return !isManualBaudRate() && !Platform_CommSharingWithApplication();
+}
+
+int Platform_CommIsWaitingForGdbToConnect(void)
+{
+    //if (!Platform_CommShouldWaitForGdbConnect())
+    	return 0;//stm32f429 does not support auto-baudrate
+}
+
+
+
+void Platform_CommPrepareToWaitForGdbConnection(void)
+{
+    //if (!Platform_CommShouldWaitForGdbConnect())
+        return;//Due to the Platform_CommShouldWaitForGdbConnect() always return 0,the if condition always enter!
+}
+
+
+static int hasHostSentDataInLessThan10Milliseconds(void);
+static int isNoReceivedCharAndNoTimeout(void);
+
+void Platform_CommWaitForReceiveDataToStop(void)

```

Comment:
Since you aren't using auto-baud, you can probably just have this do nothing and then you would have no code which twiddles with the SysTicker.